packages/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -1,26 +1,245 @@
+    fontSize: '10px',
+    width: '10px',
+    height: '10px',
+  },
+
+  // TODO: Remove fill once checkbox uses react-icons
+  disabled: theme => ({
+    opacity: 1,
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+    fill: theme.alias.color.neutral.neutralForegroundDisabled,
+  }),
+
+  unchecked: {
+    opacity: 0,
+  },
+
+  // TODO: neutralForegroundInverted change to NeutralForegroundOnBrand once it's added
+  checked: theme => ({
+    opacity: 1,
+    fill: theme.alias.color.neutral.neutralForegroundInverted,
+  }),
+
+  mixed: theme => ({
+    opacity: 1,
+    fill: theme.alias.color.neutral.compoundBrandForeground1,
+    ':hover': {
+      fill: theme.alias.color.neutral.compoundBrandForeground1Hover,
+    },
+    ':active': {
+      fill: theme.alias.color.neutral.compoundBrandForeground1Pressed,
+    },
+  }),
+});
+
+const useLabelStyles = makeStyles({
+  label: {
+    userSelect: 'none',
+    cursor: 'pointer',
+  },
+
+  disabled: theme => ({
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+    cursor: 'default',
+  }),
+
+  unchecked: theme => ({
+    color: theme.alias.color.neutral.neutralForeground3,
+    ':hover': {
+      color: theme.alias.color.neutral.neutralForeground2,
+    },
+    ':active': {
+      color: theme.alias.color.neutral.neutralForeground1,
+    },
+  }),
+
+  checked: theme => ({
+    color: theme.alias.color.neutral.neutralForeground1,
+  }),
+
+  mixed: theme => ({
+    color: theme.alias.color.neutral.neutralForeground1,
+  }),
 });
 
 /**
  * Apply styling to the Checkbox slots based on the state
  */
 export const useCheckboxStyles = (state: CheckboxState): CheckboxState => {
+  const checkedState = state.checked === 'mixed' ? 'mixed' : state.checked ? 'checked' : 'unchecked';
+  const indicatorStyles = useIndicatorStyles();
+  const labelStyles = useLabelStyles();
+  const inputStyles = useInputStyle();
+  const boxStyles = useBoxStyles();
   const styles = useStyles();
-  state.className = mergeClasses(styles.root, state.className);
 
-  // TODO Add class names to slots, for example:
-  // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);
+  state.className = mergeClasses(styles.root, styles.focusIndictor, state.className);
+
+  state.input.className = mergeClasses(
+    boxStyles[state.size],
+    inputStyles.input,
+    state.disabled && inputStyles.disabled,
+    state.input.className,
+  );
+
+  state.checkboxClassName = mergeClasses(
+    boxStyles.box,
+    boxStyles[state.size],
+    !!state.label.children && boxStyles[state.labelPosition],
+    !state.disabled && boxStyles[checkedState],
+    state.disabled && boxStyles.disabled,
+    state.circular && boxStyles.circular,
+  );
+
+  state.indicator.className = mergeClasses(
+    indicatorStyles[state.size],
+    indicatorStyles.indicator,
+    state.disabled && indicatorStyles.disabled,
+    !state.disabled && indicatorStyles[checkedState],
+    state.indicator.className,
+  );
```

Comment:
I think we'd want the `indicator` slot to get the "box" styles. The entire indicator slot will consist of the box, with the checkmark icon as its child. That will let users customize the styles of the box as well.

The div that currently gets checkboxClassName would instead just be a container to overlap the `indicator` and `input` slots. And then combine `boxStyles` and `indicatorStyles` into one, since they'd both be applied to the same div.

(Feel free to address this comment in a future PR. It doesn't need to block this PR)